### PR TITLE
script: implement `import.meta.resolve`

### PR DESCRIPTION
**Pull request by Gae24**
Our implementation of [HostGetImportMetaProperties](https://html.spec.whatwg.org/multipage/#hostgetimportmetaproperties) wasn't providing `resolve` function to mozjs. 

Testing: Covered by existing tests, results updated
Part of #37553

### CODE REVIEW

File: tests/wpt/meta/import-maps/data-driven/resolving.html.ini [hunk 1]
```diff
@@ -1,551 +1,45 @@
...
-[resolving.html?data-url-prefix.json]
 
 [resolving.html?empty-scopes.json]
-  [Scope sorting with multiple empty scopes: Multiple empty scopes before specific scope: Import from foo should use version2: bar]
-    expected: FAIL
-
-  [Scope sorting with multiple empty scopes: Multiple empty scopes before specific scope: Import from root bundles should use version1: bar]
-    expected: FAIL
-
   [Scope sorting with multiple empty scopes: Multiple empty scopes before specific scope: Import from asdfasdf should use version1 (empty scope): bar]
```
Comment (TimvdLippe) [inline tests/wpt/meta/import-maps/data-driven/resolving.html.ini:41]: Is there any particular reason we are still failing those? Are those for other reasons than the current implementation?

**Review by TimvdLippe — COMMENTED** [review]
I am not an expert in mozjs side of things, but given the WPT results I think it's fine? Asked one question about the remaining test failures whether we can also address these in this PR or require more follow-up work.

File: tests/wpt/meta/import-maps/data-driven/resolving.html.ini (hunk 1 above)
Comment (Gae24) [reply tests/wpt/meta/import-maps/data-driven/resolving.html.ini:41]: I've not investigated it yet, but it's probably due to some corner case of [resolve a module specifier](https://html.spec.whatwg.org/multipage/#resolve-a-module-specifier) that we are missing.

**Review by TimvdLippe — APPROVED** [review]
The code looks okay to me and given so many WPT tests start to pass, let's merge this. Do try to figure out in a follow-up how to fix the remaining test failures. Maybe there is an underlying issue that we can resolve (no pun intended).
